lib/widgets/info_card.dart

```diff
@@ -18,85 +19,49 @@ class InfoCard<T> extends StatelessWidget {
-                            color: Theme.of(context).primaryColor,
-                            size:
-                                Theme.of(context).textTheme.subtitle2.fontSize,
-                          )
-                        ],
-                      ),
-                    ),
-                ],
-              ),
-              const SizedBox(height: 10),
-              FutureBuilder(
-                  future: future,
-                  builder: (final context, final snapshot) {
-                    if (snapshot.connectionState == ConnectionState.done) {
-                      if (snapshot.hasData) {
-                        if ((snapshot.data is Map ||
-                                snapshot.data is Iterable) &&
-                            snapshot.data.isEmpty) {
-                          return noneYet(context);
-                        }
+    return FutureBuilder(
+        future: future,
+        builder: (final context, final snapshot) {
+          if (snapshot.connectionState == ConnectionState.done) {
+            // this condition resolves to true if the snapshot contains some data and if that data is iterable and not empty
+            final bool toDisplayContent = snapshot.hasData &&
+                !((snapshot.data is Map || snapshot.data is Iterable) &&
+                    snapshot.data.isEmpty);
 
-                        return builder(snapshot.data);
-                      } else {
-                        return noneYet(context);
-                      }
-                    }
-                    return const SizedBox(
-                      height: 100,
-                      child: Center(child: CircularProgressIndicator()),
-                    );
-                  }),
-            ],
-          ),
-        ),
-      ),
-    );
+            // if there is no content to be displayed, we default to the chosen strategy for the no-items widget
+            return toDisplayContent
+                ? cardWrapper(
+                    context: context, cardContent: builder(snapshot.data))
+                : (showIfEmpty
+                    ? cardWrapper(
+                        context: context, cardContent: noneYet(context))
+                    : const SizedBox());
+          }
+          return const SizedBox(
+            height: 100,
+            child: Center(child: CircularProgressIndicator()),
+          );
+        });
   }
 
+  Widget cardWrapper(
+          {@required final BuildContext context,
+          @required final Widget cardContent}) =>
+      Padding(
+          padding: padding ?? const EdgeInsets.fromLTRB(12, 12, 12, 0),
+          child: Card(
+              child: Padding(
+                  padding: const EdgeInsets.all(12),
+                  child: Column(children: <Widget>[
+                    cardHeader(context),
+                    const SizedBox(height: 10),
+                    cardContent,
+                  ]))));
```

Comment:
Add trailing commas after every `)`. It improves the code style.

Reply:
done, Android Studio makes it more readable afterwards to see the end of a widget